ledger/streaming/client/src/grpc.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use displaydoc::Display;
+use futures::{Stream, StreamExt};
+use mc_common::logger::{log, o, Logger};
+use mc_crypto_keys::Ed25519Public;
+use mc_ledger_streaming_api::{
+    parse_subscribe_response,
+    streaming_blocks::{SubscribeRequest, SubscribeResponse},
+    streaming_blocks_grpc::LedgerUpdatesClient,
+    BlockStream, BlockStreamComponents, Result,
+};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConnectionUri;
+use std::{sync::Arc, time::Duration};
+
+/// A [BlockStream] that streams blocks using the `LedgerUpdates` gRPC API.
+#[derive(Display)]
+pub struct GrpcBlockSource {
+    /// The gRPC client
+    client: LedgerUpdatesClient,
+    /// The public key of the consensus node we're streaming from.
+    public_key: Arc<Ed25519Public>,
```

Comment:
This component shouldn't be checking signatures on blocks it receives, there should be a downstream object which is doing that. The only thing this object should be doing is handling the gRPC to get the blocks.

Reply:
Currently the validator does check the sigs & the SCP component checks quorums/combines based SCP rules

Reply:
I see what you're getting at, but there's currently no way to propagate the `SubscribeResponse.result_signature`, so we basically have to verify when we convert from `SubscribeResponse` to `BlockStreamComponents`. I'm up for splitting this sink into an element that basically maps gRPC->local and another that converts `SubscribeResponse` to `BlockStreamComponents`, in a follow-up PR.